### ISSUE TITLE
Warlock's Problems 

### ISSUE THREAD

**Issue by Brittany98** [SunwellTracker/issues#949]
Decription: 
1- The Egg of Mortal Essence Doesn't work. 
2- Soul Shatter takes is not scripted well.
3- Shadow Cleave (Metamorphosis) is not scripted well.
How it works: 
1- Warlock's Pet Synergy from the Talent point in Demonology doesn't seem to proc it at all which is completely incorrect.

2-Soul Shatter takes 2 seconds to work after using it.

3- It seems that when I metamorphosis and spam shadow cleave while casting a spell, Shadow cleave is casted {highlighted until cast ends} and the next press on it, it stops {unhighlighted}, which it shouldn't. It is like warriors can not spam Heroic strike which is completely incorrect. Plus, when i'm casting and its highlighted, it doesn't go off after the cast, my character doesn't use shadow cleave after cast is done. 

How it should work: 
1- Warlock's Pet Synergy from the Talent point in Demonology tree should proc it all the time. A proof of screen shot is included.

2- Soul Shatter should lower the threat instantly not 2 seconds later. With 2 seconds later, warlocks technically have to pre-soul shatter threat in order to survive. 

3- In retail wow the way it works is as follows:
When it is being spammed while casting another spell, it should remain highlighted {pressed/casted} all the time. It is not supposed to be highlighted when I first press it and unhighlighted when I press it the second time.

Second, it should instantly goes off right after cast. But the way it works here is that you have to wait for a couple of seconds until it goes off.

Source (you should point out proofs of your report, please give us some source):

**Comment by Akha72** [comment]
I totally agree with point number 3 - it seems as if for shadow cleave to work a warlock has to stop casting which leads to a dps drop and as a consequence players omitting shadow cleave in his/her rotation completely. I will look for some retail video with demo pve in my spare time.

**Comment by Cwiercpasiec** [comment]
Point 1.
@Brittany98 Any source? And by source I mean proofs from global. Here is something i found: http://www.wowhead.com/forums&topic=114333/the-egg-of-mortal-essence
Discusion from patch 3.1.3, according to this thread, the only thing that proced this trinket was Fel Armor, and the last comment states "Apparently it no longer procs off Fel Armor heals.".

**Comment by Brittany98** [comment]
@Cwiercpasiec Concerning the first point, back in retail, people had messed up talent points. In that discussion, there did not mention Fel Synergy at all which sounds weird. Fel Synergy directly heals your pet when you hit the target, and thus, it should proc it because Egg is proced by any direct healing as it states in the trinket it self. I included a screen shot of my warlock in Dalaran-wow and I never had post it there in the bug discussion because they knew it should work for warlocks. Also, I myself played retail and I used to have this trinket and embrace of the spider in my PVP spec and it worked well but unfortunately, I can't go back in time and get a screen shot. But, people with no experience in WOTLK does not try to take advantage of every single spell/talent point, and that is why I believe you think it should only work for healers. 

Concerning the second point, i am not talking about the visuals it has when you use it, I am talking about having the effect of reducing your threat. Go to you interface>Display>Show aggro, and try to have high threat and use soul shatter, your aggro will drop 2-3 seconds later not instantly. The way it worked in retail, once you over aggro the boss and use soul shatter, your aggro drops right away and the boss goes back to the second highest aggro which is 99% of the time is the tank.

**Comment by Cwiercpasiec** [comment]
I asked for source, because noone will change the way a trinket procs solely cause someone said it worked this way on retail. I never played on retail and all I can do now is search for some sources, proving or disproving your report. I found some discussions about Egg of Mortal Essence procing for warlocks, some people said it proced from Drain Life, some said it was from Fel Armor heal, and some mentioned Life Tap, but I haven't seen anyone mentioning that this trinket could proc from Fel Synergy. 
Here: https://www.mmo-champion.com/threads/668574-Healing-trinket-proc-for-warlocks
This is the only place I could find someone write anything about Fel Synergy regarding the Egg, but noone confirmed nor denied. By the way, tooltip is not a source, nor are screenshots from other private servers. 
I don't really think that if Egg could proc from Fel Synergy noone would mention it anywhere, but you can prove me wrong by simply finding a source.

Regarding the second point, I've spoke with a developer. He told me, that the server sends updated threat table to client every 2 seconds and that's the reason why you see the effect of Soul Shatter quite delayed. This doesn't affect NPC though, because server checks if the target is the one that should be attacked every 0,1 second (or something like that, I don't want to lie and don't remember exact number, but it was something close to 0,1s). So basically you just see Soul Shatter taking effect with a delay, but it actually works instantly.
And by the way, using threat reducing skills after over aggroing a boss is very bad habbit.
